common/src/main/scala/uk/ac/wellcome/s3/S3ObjectLocation.scala

```diff
@@ -0,0 +1,20 @@
+package uk.ac.wellcome.s3
+
+import java.net.URI
+
+import scala.util.Try
+
+case class S3ObjectLocation(bucket: String, key: String) {
+  def toURI: URI = {
+    new URI(s"s3://$bucket/$key")
+  }
+}
+
+object S3ObjectLocation {
+  def create(uri: URI): Try[S3ObjectLocation] =
```

Comment:
This is also never used except for the `create` method in `MessagePointer` which is not used